benchopt/base.py

```diff
@@ -392,7 +392,11 @@ def set_dataset(self, dataset):
             parameters[key] = getattr(self, key)
         self.set_data(**data)
         for key in self._parameters:
-            if parameters[key] != getattr(self, key):
+            has_changed = (any(parameters[key] != getattr(self, key))
+                           if hasattr(parameters[key], '__iter__')
+                           else parameters[key] != getattr(self, key))
```

Comment:
can you give an example of parameter value that creates a pb?

Reply:
yes sure

```python
# ...
    parameters = {
        'weights': [np.ones(n_groups), abs(np.random.randn(n_groups))],   
# ...
```

Reply:
I would use 

# ...
    parameters = {
        'weights': ["ones", "randn"],   
# ...

and do the set in the run function. Also it will allow to fix the seed of randn

Reply:
Agree!
but what about

```python
# ...
    parameters = {
        'groups': [[5, 7, 3], [5, 5, 2, 3]], # contiguous groups of 15 features
# ...
```

Anyways, I think the case of iterable params arises rarely. So no need to handle it at all.